### ISSUE TITLE
Simplify simplify simplify

### ISSUE THREAD

**Issue by Azeirah** [Azeirah/nv#8]
@HeitorAugustoLN I'm reading the flake.nix now, and I honestly merged your contribution without even looking at it because I was happy to get a contribution in the first place 😂

But, I think the flake is a little overcomplicated for what it is. The primary goals is to have something that is easy, understandable, and works out of the box, but most importantly it is extensible.

I'm just reading through the flake from top to bottom, and commenting what I think in this thread as I go through it.

**Comment by Azeirah** [comment]
> https://github.com/Azeirah/nv/blob/fabef5de8bb53dd36431d04a767dee43ff2b1160/flake.nix#L52

I would say we pin to 24.11? Because then everybody is always on the same version of everything at the start. The comment for this line can include 

```
# you can switch to nixos-unstable by typing nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
```

**Comment by Azeirah** [comment]
> https://github.com/Azeirah/nv/blob/fabef5de8bb53dd36431d04a767dee43ff2b1160/flake.nix#L36

I don't really know if we need nightly? Basic nixvim is fine. People can always upgrade if they want. I want the basics as basic as they can be/

This part, the inputs can be just this, right?

```
inputs = {
  nixvim.url = "github:nix-community/nixvim";
}
```

And is treefmt necessary as an explicit input? Nixvim has it as a default, right? Why override the default with itself?

I see this comment in nixvim main repository:

```
    /*
      # NOTE: The inputs below this comment are optional
      # You can remove them with `inputs.<input>.follows = ""`

      For example:

      ```
      nixvim = {
        url = "github:nix-community/nixvim";
        inputs = {
          nixpkgs.follows = "nixpkgs";
          flake-parts.follows = "flake-parts";

          devshell.follows = "";
          flake-compat.follows = "";
          git-hooks.follows = "";
          home-manager.follows = "";
          nix-darwin.follows = "";
          treefmt-nix.follows = "";
        };
      };
      ```
    */
```

We can always just copy and paste that comment and allow people to remove optional dependencies.

**Comment by Azeirah** [comment]
https://github.com/Azeirah/nv/blob/fabef5de8bb53dd36431d04a767dee43ff2b1160/flake.nix#L90C1-L96C5

Cachix is probably god! But we should explain what cachix _adds_ here, why is it good/helpful? Is it necessary? And where does that key come from?

**Comment by Azeirah** [comment]
>     flake-parts.url = "github:hercules-ci/flake-parts";

Also, this. I'm not super sure what the value proposition is for this. I usually use the flake-utils thing, but that's just because everyone else uses it. Does flake-parts offer anything extra for a super basic extensible nix nvim editor? If so? Great! We should keep it, but we need to explain why we're using it, and where it provides benefit.

**Comment by Azeirah** [comment]
https://github.com/Azeirah/nv/blob/fabef5de8bb53dd36431d04a767dee43ff2b1160/flake.nix#L135C1-L139C13

This is quite nice. I love having a `checks` thing, it's nice to show that "testing/checking" is built-in to nix flakes.

**Comment by Azeirah** [comment]
https://github.com/Azeirah/nv/blob/fabef5de8bb53dd36431d04a767dee43ff2b1160/flake.nix#L141C1-L155C1

I'm not super sure, why do we need a dev version of this? That's not asking as like "can we get rid of this?", I mean I genuinely don't see why we need a dev shell? Does that provide any particular benefit?

Also, if you ask me the nightly can just be dropped, focus on the essentials.

**Comment by Azeirah** [comment]
https://github.com/Azeirah/nv/blob/fabef5de8bb53dd36431d04a767dee43ff2b1160/flake.nix#L157C1-L166C13

This part is good too. Though, again, I think the nightly is not necessary.

**Comment by Azeirah** [comment]
@HeitorAugustoLN What are your thoughts?

**Comment by HeitorAugustoLN** [comment]
I agree that most of these aren't necessary for a starting point, I added them since I thought `nv` would be a distro. But I still think adding neovim nightly could be useful.

> And is treefmt necessary as an explicit input?

I added it for formatting the nix and lua files in the repo, though we could use the normal version too

> Is it necessary? And where does that key come from?

Not really, it is just to have a binary cache for neovim nightly, instead of having to build it. The key comes from: https://app.cachix.org/cache/nix-community
